tools/check-enforcer/Azure.Sdk.Tools.CheckEnforcer/Handlers/CheckRunHandler.cs

```diff
@@ -1,23 +1,63 @@
 using System;
 using System.Collections.Concurrent;
 using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
 using System.Runtime.CompilerServices;
 using System.Text;
 using System.Threading;
 using System.Threading.Tasks;
+using YamlDotNet.Core;
 
 namespace Azure.Sdk.Tools.CheckEnforcer.Handlers
 {
     public class CheckRunHandler : Handler<CheckRunEventPayload>
     {
-        public CheckRunHandler(IGlobalConfigurationProvider globalConfigurationProvider, IGitHubClientProvider gitHubCLientProvider, IRepositoryConfigurationProvider repositoryConfigurationProvider, ILogger logger) : base(globalConfigurationProvider, gitHubCLientProvider, repositoryConfigurationProvider, logger)
+        public CheckRunHandler(IGlobalConfigurationProvider globalConfigurationProvider, IGitHubClientProvider gitHubCLientProvider, IRepositoryConfigurationProvider repositoryConfigurationProvider, ILogger<CheckRunHandler> logger) : base(globalConfigurationProvider, gitHubCLientProvider, repositoryConfigurationProvider, logger)
         {
         }
 
+        public override string EventName => "check_run";
+
+        protected override IEnumerable<CheckRunEventPayload> FilterPayloads(IEnumerable<CheckRunEventPayload> payloads)
+        {
+            // TODO: This filtering logic needs to be rationalized with the HandleCoreAsync
+            //       method, as it duplicates the filtering we do there.
+
+            // If a user explicitly requests an evaluation via the evaluate button, do it.
+            var requestedActionPayloads = from payload in payloads
+                                          where payload.CheckRun.Name == this.GlobalConfigurationProvider.GetApplicationName()
+                                            && payload.Action == "requested_action"
+                                            && payload.RequestedAction.Identifier == "evaluate"
+                                          select payload;
+
+            // Here we just want to take completed events since nothing else matters, and
+            // only the last one since the evaluation will be the same.
+            var elligiblePayloads = from payload in payloads
+                                    where payload.CheckRun.Name != this.GlobalConfigurationProvider.GetApplicationName()
+                                    where payload.CheckRun.StartedAt < DateTimeOffset.UtcNow.Subtract(TimeSpan.FromDays(1))
+                                    where payload.CheckRun.Status == new StringEnum<CheckStatus>(CheckStatus.Completed)
+                                    select payload;
+
+            var everythingElse = from payload in elligiblePayloads
+                                 group payload by payload.CheckRun.HeadSha into payloadGroup
+                                 select payloadGroup.Last();
+
+            var delta = elligiblePayloads.Count() - everythingElse.Count();
+
+            if (delta > 0)
+            {
+                Debugger.Break();
```

Comment:
Focusing on this delta I think is the most interesting thing. Are we hitting this often enough or do we really need to focus on caching the identical API calls themselves (and off box?).
[A drop in distributed cache](https://docs.microsoft.com/en-us/aspnet/core/performance/caching/distributed?view=aspnetcore-3.1#distributed-redis-cache) might be easy enough.